homeassistant/components/reolink/sensor.py

```diff
@@ -0,0 +1,121 @@
+"""Component providing support for Reolink sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import date, datetime
+from decimal import Decimal
+
+from reolink_aio.api import Host
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkHostCoordinatorEntity
+
+
+@dataclass
+class ReolinkHostSensorEntityDescriptionMixin:
+    """Mixin values for Reolink host sensor entities."""
+
+    value: Callable[[Host], bool]
+
+
+@dataclass
+class ReolinkHostSensorEntityDescription(
+    SensorEntityDescription, ReolinkHostSensorEntityDescriptionMixin
+):
+    """A class that describes host sensor entities."""
+
+    supported: Callable[[Host], bool] = lambda host: True
+
+
+HOST_SENSORS = (
+    ReolinkHostSensorEntityDescription(
```

Comment:
Why not set this directly on the `ReolinkHostSensorEntity` as you do with `EventConnectionSensorEntity`

Reply:
Because EventConnectionSensorEntity is a weird one that does not get its state directly form the API but from the host class living inside the HomeAssistant code.

All other reolink entities get there state from the API directly and are defined in this way. If I want to add aditional sensors I just need to add the entity description to the list of HOST_SENSORS and done. So this is for future proofing.
See for example the switch platform of Reolink:
https://github.com/home-assistant/core/blob/47952be112554869bcb780ac1bb68df63f1c2c23/homeassistant/components/reolink/switch.py#L55-L201

Reply:
Sure. My comment was more based that it looked like a one-off but let's keep it as is.